src/presets/presetsController.ts

```diff
@@ -304,6 +306,32 @@ export class PresetsController implements vscode.Disposable {
                         if (chosen_kit.kit.name === SpecialKits.ScanForKits) {
                             await KitsController.scanForKits(await this.project.getCMakePathofProject());
                             return false;
+                        } else if (chosen_kit.kit.name === SpecialKits.ScanSpecificDir) {
+                            const dir = await vscode.window.showOpenDialog({
+                                canSelectFiles: false,
+                                canSelectFolders: true,
+                                canSelectMany: false,
+                                openLabel: localize('select.folder', 'Select Folder')
+                            });
+                            if (!dir || dir.length === 0) {
+                                return false;
+                            }
+                            const dirPathWithDepth = async (folder: string, depth: number = 5) => {
```

Comment:
I'm noticing that thsi code is duplicated in the `kitsController.ts` and the `presetsController.ts`. Please extract it into a method so that we share code. Likely the best place to put it is in the `kitsController` alongside `scanForKits` or in `kit.ts`.

Reply:
Extracted into shared method.